app/helpers/users_helper.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module UsersHelper
+  def allowed_user?(user)
+    current_user.id == user.id
```

Comment:
```suggestion
    current_user == user
```

だけでも大丈夫です。

ただ、これだとあまりメソッド化するメリットはないかも？
https://bootcamp.fjord.jp/pages/when-you-should-define-method